web-marketplace/src/pages/EcocreditsByAccount/hooks/useMergedCreditClasses.tsx

```diff
@@ -0,0 +1,91 @@
+import { useMemo } from 'react';
+
+import { useCreditClasses } from 'pages/Home/hooks/useCreditClasses';
+import { useClassesWithMetadata } from 'hooks/classes/useClassesWithMetadata';
+import { useFetchCreditClassesWithOrder } from 'hooks/classes/useFetchCreditClassesWithOrder';
+
+export const useMergedCreditClasses = (adminId?: string) => {
+  const {
+    creditClasses: adminClasses,
+    isLoadingCreditClasses: adminClassesLoading,
+  } = useFetchCreditClassesWithOrder({ admin: adminId });
+
+  const {
+    creditClasses: sanityCreditClasses,
+    creditClassesPrograms,
+    loading: sanityCreditClassesLoading,
+  } = useCreditClasses({});
+
+  const classesMissingSanityData = useMemo(() => {
+    if (!adminClasses?.length) return [];
+    return adminClasses.filter(
+      adminClass => !sanityCreditClasses?.some(sc => sc.path === adminClass.id),
+    );
+  }, [adminClasses, sanityCreditClasses]);
+
+  const { classesMetadata, isClassesMetadataLoading } = useClassesWithMetadata(
+    classesMissingSanityData.map(c => c.id),
+  );
+
+  const metadataByClassId = useMemo(() => {
+    const map = new Map();
+    classesMissingSanityData.forEach((missingClass, index) => {
+      if (classesMetadata?.[index]) {
+        map.set(missingClass.id, classesMetadata[index]);
+      }
+    });
+    return map;
+  }, [classesMissingSanityData, classesMetadata]);
+
+  const hasSanityContent = (sanityClass: any) =>
+    sanityClass?.shortDescriptionRaw || sanityClass?.descriptionRaw;
```

Comment:
do we really need this fn? it's used only below
also if we keep it let's avoid any